packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,311 @@
+<template>
+  <KPop
+    class="analytics-timepicker"
+    placement="bottomStart"
+    width="auto"
+    hide-caret
+  >
+    <KButton
+      :is-rounded="false"
+      size="large"
+      class="timepicker-input"
+    >
+      {{ displayedRange }}
+    </KButton>
+    <template #content>
+      <KSegmentedControl
+        v-if="hasRelativeTimeframes"
+        v-model="mode"
+        :options="[
+          { label: 'Custom', value: 'custom' },
+          { label: 'Relative', value: 'relative' }
+        ]"
+        class="w-100 mb-5"
+        data-testid="analytics-time-toggle"
+      />
+      <DatePicker
+        v-if="showCalendar"
+        :attributes="attrs"
+        :model-config="modelConfig"
+        v-model="selectedCalendarRange"
+        mode="dateTime"
+        is-range
+        is-expanded
+      />
+      <div
+        v-else-if="!showCalendar && hasRelativeTimeframes"
```

Comment:
`v-else-if` already rules out `!showCalendar`, so we can just
```suggestion
        v-else-if="hasRelativeTimeframes"
```

Reply:
You make an excellent point here 🤣